### PR TITLE
adds cassandra-medusa image

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
### Image Size

- [x] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes:

### Image Vulnerabilities

- [x] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Image Tagging

- [ ] The image is _not_ tagged with version tags.
- [x] The image is tagged with :latest
- [ ] The image is not tagged with :latest (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [x] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes:

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [x] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [x] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [ ] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes: Might have to test this with AWS too, will probably need access and not sure if that can be automated

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [ ] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [x] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

Notes: From the dockerfile, it required GID/UID as 999 with cassandra username https://github.com/thelastpickle/cassandra-medusa/blob/c8609c8eaeeca6bf05f46fd50145eed2b3ae5a3d/k8s/Dockerfile#L53

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [x] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [x] Environment variables added.
- [ ] Environment variables not added and not required.

### SIGTERM

- [x] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [x] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

File: images/cassandra-medusa/tests/main.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  repository       = "https://charts.jetstack.io"
+  chart            = "cert-manager"
+  create_namespace = true
+
+  set {
+    name  = "installCRDs"
+    value = "true"
+  }
+}
+
```
Comment (mamccorm) [inline images/cassandra-medusa/tests/main.tf:29]: This is looking great so far. I think we could further extend the tests here by doing some log validation after deploying via helm.

Some examples from our grafana image:

- https://github.com/chainguard-images/images/blob/03632b6ed8e41ee3fd121eca2dec96733c0f10d8/images/grafana/tests/main.tf#L33
- https://github.com/chainguard-images/images/blob/03632b6ed8e41ee3fd121eca2dec96733c0f10d8/images/grafana/tests/smoke-test.sh

Comment (mritunjaysharma394) [reply images/cassandra-medusa/tests/main.tf:29]: Thanks @mamccorm , I guess these logs based test might not gave great results as its a CRD based tests and we might need to create objects related to medusa from CRD, only then we can verify it to that level. For that, we will have to use AWS and probably will need a S3 bucket, for ref: https://docs.k8ssandra.io/tasks/backup-restore/

File: images/cassandra-medusa/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Medusa for Apache Cassandra® is deployed by a K8ssandra Operator install, based on the Medusa Custom Resource Definition (CRD). Once K8ssandra Operato is deployed, you can refer to the [official documentation](https://docs.k8ssandra.io/tasks/backup-restore/) for further usage of Medusa 
+
+To use our minimal, wolfi-based image with this Helm chart you'll need to override the image used by the official helm chart and specify the chainguard image as per below example:
+
+```shell
+helm repo add k8ssandra https://helm.k8ssandra.io/stable
+helm repo update
+
+helm install cassandra-medusa k8ssandra/k8ssandra-operator -n cassandra-medusa
+helm upgrade k8ssandra-operator \
```
Comment (mamccorm) [inline images/cassandra-medusa/README.md:40]:
```suggestion
helm install k8ssandra-operator \
```

File: images/cassandra-medusa/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```
+docker pull cgr.dev/chainguard/cassandra-medusa:latest
+```
+<!--getting:end-->
+
+<!--body:start-->
+
+## Usage
+Medusa for Apache Cassandra® is deployed by a K8ssandra Operator install, based on the Medusa Custom Resource Definition (CRD). Once K8ssandra Operato is deployed, you can refer to the [official documentation](https://docs.k8ssandra.io/tasks/backup-restore/) for further usage of Medusa
```
Comment (mamccorm) [inline images/cassandra-medusa/README.md:31]:
```suggestion
Medusa for Apache Cassandra® is deployed by a K8ssandra Operator install, based on the Medusa Custom Resource Definition (CRD). Once K8ssandra Operator is deployed, you can refer to the [official documentation](https://docs.k8ssandra.io/tasks/backup-restore/) for further usage of Medusa 
```

File: images/cassandra-medusa/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+To use our minimal, wolfi-based image with this Helm chart you'll need to override the image used by the official helm chart and specify the chainguard image as per below example:
+
+```shell
+helm repo add k8ssandra https://helm.k8ssandra.io/stable
+helm repo update
+
+helm install cassandra-medusa k8ssandra/k8ssandra-operator -n cassandra-medusa
+helm upgrade k8ssandra-operator \
+    -n cassandra-medusa \
+    --set medusa.image.registry=cgr \
```
Comment (mamccorm) [inline images/cassandra-medusa/README.md:42]:
```suggestion
kubectl create ns cassandra-medusa
helm install k8ssandra-operator \
    -n cassandra-medusa \
    --set medusa.image.registry=cgr \
```

File: images/cassandra-medusa/tests/main.tf (hunk 1 above)
Comment (mamccorm) [reply images/cassandra-medusa/tests/main.tf:29]: Thanks, yea looks like the operator only creates a pod for medusa whenever it need to run a backup job.

I was wondering can we at least force the operator to create the pod in the test, even if the backup itself fails.

i.e:

```bash
apiVersion: k8ssandra.io/v1
kind: K8ssandraCluster
metadata:
  name: my-k8ssandra-cluster
  namespace: cassandra-medusa
spec:
  cassandra:
    cluster: my-cassandra-cluster
    datacenters:
    - metadata:
        name: dc1
      size: 3
```

But I am not really sure how to deploy this. looks like its looking for CRDs now

**Review by mamccorm — APPROVED** [review]

**Pull request merged**